File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.io.IOException;
import java.util.*;

class Solution {
     boolean solution(String s) {
        boolean answer = true;

        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            char element = s.charAt(i);

            if (element == '(') {
                stack.push(element);
            }
            else {
                if (stack.size() == 0) {
                    return false;
                } else stack.pop();
            }
        }
        
        if (stack.isEmpty()) {
            return answer;
        }
        return false;
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    
    for t in list(s):
        if(t == '('):
            stack.append('(')
        else:
            if(len(stack) == 0):
                return False
            else:
                stack.pop()
    if(len(stack) > 0):
        return False
    return True
